### ISSUE TITLE
netifaces recipe broken

### ISSUE THREAD

**Issue by brussee** [kivy/python-for-android#802]
Regression introduced in #783 

```
[INFO]:    Building netifaces for armeabi-v7a
('name is', 'netifaces', <class 'pythonforandroid.recipes.netifaces.NetifacesRecipe'>)
[INFO]:    netifaces apparently isn't already in site-packages
[INFO]:    Building compiled components in netifaces
ccache found, will optimize builds
('path is', '/home/paul/android-ndk-r10e/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86/bin/:/home/paul/android-ndk-r10e/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/:/home/paul/android-ndk-r10e:/home/paul/android-sdk-linux/tools:~/.local/bin/:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/home/paul/.local/bin:/opt/android-sdk-linux/platform-tools:/opt/android-ndk-r10e')
[INFO]:    -> directory context /home/paul/.local/share/python-for-android/build/other_builds/netifaces/armeabi-v7a/netifaces
[DEBUG]:   -> running hostpython setup.py build_ext -v
[DEBUG]:    running build_ext
[DEBUG]:    checking for getifaddrs...not found. (cached)
[DEBUG]:    checking for getnameinfo...found. (cached)
[DEBUG]:    checking for socket IOCTLs...['SIOCGIFCONF', 'SIOCGIFHWADDR', 'SIOCGIFADDR', 'SIOCGIFFLAGS', 'SIOCGIFDSTADDR', 'SIOCGIFBRDADDR', 'SIOCGIFNETMASK']. (cached)
[DEBUG]:    checking for optional header files...netpacket/packet.h linux/atm.h. (cached)
[DEBUG]:    checking whether struct sockaddr has a length field...no. (cached)
[DEBUG]:    checking which sockaddr_xxx structs are defined...in in6 un ll atmpvc atmsvc. (cached)
[DEBUG]:    checking for routing socket support...no. (cached)
[DEBUG]:    checking for sysctl(CTL_NET...) support...no. (cached)
[DEBUG]:    checking for netlink support...yes. (cached)
[DEBUG]:    will use netlink to read routing table
[DEBUG]:    building 'netifaces' extension
[DEBUG]:    /usr/bin/ccache arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /home/paul/android-ndk-r10e/platforms/android-16/arch-arm -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DANDROID -mandroid -fomit-frame-pointer --sysroot /home/paul/android-ndk-r10e/platforms/android-16/arch-arm -march=armv7-a -mfloat-abi=softfp -mfpu=vfp -mthumb -fPIC -DNETIFACES_VERSION=0.10.4 -DHAVE_GETNAMEINFO=1 -DHAVE_SIOCGIFCONF=1 -DHAVE_SIOCGIFHWADDR=1 -DHAVE_SIOCGIFADDR=1 -DHAVE_SIOCGIFFLAGS=1 -DHAVE_SIOCGIFDSTADDR=1 -DHAVE_SIOCGIFBRDADDR=1 -DHAVE_SIOCGIFNETMASK=1 -DHAVE_SOCKET_IOCTLS=1 -DHAVE_NETPACKET_PACKET_H=1 -DHAVE_LINUX_ATM_H=1 -DHAVE_SOCKADDR_IN=1 -DHAVE_SOCKADDR_IN6=1 -DHAVE_SOCKADDR_UN=1 -DHAVE_SOCKADDR_LL=1 -DHAVE_SOCKADDR_ATMPVC=1 -DHAVE_SOCKADDR_ATMSVC=1 -DHAVE_PF_NETLINK=1 -I/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include -I/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2 -c netifaces.c -o build/temp.linux-x86_64-2.7/netifaces.o
[DEBUG]:    In file included from /home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/Python.h:58:0,
[DEBUG]:                     from netifaces.c:1:
[DEBUG]:    /home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/pyport.h:849:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
[DEBUG]:     #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
[DEBUG]:      ^
[DEBUG]:    In file included from /home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/Python.h:126:0,
[DEBUG]:                     from netifaces.c:1:
[DEBUG]:    /home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/modsupport.h:27:1: warning: 'PyArg_ParseTuple' is an unrecognized format function type [-Wformat=]
[DEBUG]:     PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...) Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3);
[DEBUG]:     ^
[DEBUG]:    netifaces.c:106:28: fatal error: linux/tipc.h: No such file or directory
[DEBUG]:     #    include <linux/tipc.h>
[DEBUG]:                                ^
[DEBUG]:    compilation terminated.
[DEBUG]:    error: command '/usr/bin/ccache' failed with exit status 1
Traceback (most recent call last):
  File "/home/paul/.local/bin/p4a", line 9, in <module>
    load_entry_point('python-for-android==0.4', 'console_scripts', 'p4a')()
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 763, in main
    ToolchainCL()
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 344, in __init__
    getattr(self, command_method_name)(unknown)
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 93, in wrapper_func
    build_dist_from_args(ctx, dist, dist_args)
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/toolchain.py", line 138, in build_dist_from_args
    build_recipes(build_order, python_modules, ctx)
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/build.py", line 561, in build_recipes
    recipe.build_arch(arch)
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/recipe.py", line 873, in build_arch
    self.build_compiled_components(arch)
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/recipe.py", line 885, in build_compiled_components
    _env=env, *self.setup_extra_args)
  File "/home/paul/.local/lib/python2.7/site-packages/pythonforandroid/logger.py", line 173, in shprint
    for line in output:
  File "/home/paul/.local/lib/python2.7/site-packages/sh.py", line 565, in next
    self.wait()
  File "/home/paul/.local/lib/python2.7/site-packages/sh.py", line 500, in wait
    self.handle_command_exit_code(exit_code)
  File "/home/paul/.local/lib/python2.7/site-packages/sh.py", line 516, in handle_command_exit_code
    raise exc(self.ran, self.process.stdout, self.process.stderr)
sh.ErrorReturnCode_1: 

  RAN: '/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/hostpython setup.py build_ext -v'

  STDOUT:
running build_ext
checking for getifaddrs...not found. (cached)
checking for getnameinfo...found. (cached)
checking for socket IOCTLs...['SIOCGIFCONF', 'SIOCGIFHWADDR', 'SIOCGIFADDR', 'SIOCGIFFLAGS', 'SIOCGIFDSTADDR', 'SIOCGIFBRDADDR', 'SIOCGIFNETMASK']. (cached)
checking for optional header files...netpacket/packet.h linux/atm.h. (cached)
checking whether struct sockaddr has a length field...no. (cached)
checking which sockaddr_xxx structs are defined...in in6 un ll atmpvc atmsvc. (cached)
checking for routing socket support...no. (cached)
checking for sysctl(CTL_NET...) support...no. (cached)
checking for netlink support...yes. (cached)
will use netlink to read routing table
building 'netifaces' extension
/usr/bin/ccache arm-linux-androideabi-gcc -DANDROID -mandroid -fomit-frame-pointer --sysroot /home/paul/android-ndk-r10e/platforms/android-16/arch-arm -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DANDROID -mandroid -fomit-frame-pointer --sysroot /home/paul/android-ndk-r10e/platforms/android-16/arch-arm -march=armv7-a -mfloat-abi=softfp -mfpu=vfp -mthumb -fPIC -DNETIFACES_VERSION=0.10.4 -DHAVE_GETNAMEINFO=1 -DHAVE_SIOCGIFCONF=1 -DHAVE_SIOCGIFHWADDR=1 -DHAVE_SIOCGIFADDR=1 -DHAVE_SIOCGIFFLAGS=1 -DHAVE_SIOCGIFDSTADDR=1 -DHAVE_SIOCGIFBRDADDR=1 -DHAVE_SIOCGIFNETMASK=1 -DHAVE_SOCKET_IOCTLS=1 -DHAVE_NETPACKET_PACKET_H=1 -DHAVE_LINUX_ATM_H=1 -DHAVE_SOCKADDR_IN=1 -DHAVE_SOCKADDR_IN6=1 -DHAVE_SOCKADDR_UN=1 -DHAVE_SOCKADDR_LL=1 -DHAVE_SOCKADDR_ATMPVC=1 -DHAVE_SOCKADDR_ATMSVC=1 -DHAVE_PF_NETLINK=1 -I/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include -I/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2 -c netifaces.c -o build/temp.linux-x86_64-2.7/netifaces.o
In file included from /home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/Python.h:58:0,
                 from netifaces.c:1:
/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/pyport.h:849:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
 #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  ^
In file included from /home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/Python.h:126:0,
                 from netifaces.c:1:
/home/paul/.local/share/python-for-android/build/other_builds/hostpython2/desktop/hostpython2/Include/modsupport.h:27:1: warning: 'PyArg_ParseTuple' is an unrecognized format function type [-Wformat=]
 PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...) Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3);
 ^
netifaces.c:106:28: fatal error: linux/tipc.h: No such file or directory
 #    include <linux/tipc.h>
                            ^
compilation terminated.
error: command '/usr/bin/ccache' failed with exit status 1


  STDERR:
```
##

**Comment by brussee** [comment]
If I set the include path like in #803 it still does not work:

```
import netifaces
exceptions.ImportError: dlopen failed: cannot locate symbol "PyExc_OSError" referenced by "/data/data/org.tribler.android/files/lib/python2.7/site-packages/netifaces.so"...
```

**Comment by brussee** [comment]
This probably can be solved by setting the linker flags properly: (in addition to my comment above)
 -L' + python_install_dir + '/lib'
 -lpython2.7

**Comment by brussee** [comment]
Fixed by #841
